language/move-prover/boogie-backend/src/boogie_wrapper.rs

```diff
@@ -357,6 +357,15 @@ impl<'env> BoogieWrapper<'env> {
         let mut at = 0;
         while let Some(cap) = VERIFICATION_DIAG_STARTS.captures(&out[at..]) {
             let inbetween = out[at..at + cap.get(0).unwrap().start()].trim();
+            at = usize::saturating_add(at, cap.get(0).unwrap().end());
+
+            let msg = cap.name("msg").unwrap().as_str();
+            if msg == "expected to fail" {
```

Comment:
We have this logic already via the OTHER_DIAG_START above, can't you just add the string there?

Reply:
I couldn't because the string "expected to fail" is part of `msg`, and cannot be found in `inbetween`. Is `inbetween` supposed to include `msg`?

Reply:
Is the overall message `assert_failed(<loc>): expected to fail`? Then I don't fully understand why this error did not show up before, because the addition of the "inbetween" check didn't not change anything to this end. In any case, I'm going to approve this now so we unblock it, but it would be good to understand what went wrong.

Reply:
It seems that `inbetween` is never `assert_failed(<loc>): ...`.

As for `$ mvp CoreAddress.move` (which has no "incorrect" test cases), no "assert_failed" is captured, and nothing is reported.

As for `$mvp --check-inconsistency CoreAddress.move`, `inbetween` is, for example:
```
Execution trace:
    output.bpl(2870,9): anon0
    output.bpl(860,17): inline$$InitVerification$0$anon0
    output.bpl(2870,9): anon0$1
    output.bpl(2017,5): inline$$CoreAddresses_TREASURY_COMPLIANCE_ADDRESS$0$anon0
    output.bpl(2898,5): anon6_Else
    output.bpl(2932,16): anon7_Then
Augmented execution trace:
$at(4,3088,3310)
...
```
This is the cause of the test failure we got this morning. If this is unexpected, perhaps, Boogie's additional output messages for the inconsistency test (e.g., "inconsistency_detected" when "assert false;" passes) have had some side effects to BoogieWrapper?

As for `$ mvp emits.move` (which has "incorrect" test cases), some "assert_failed" are captured, but `inbetween` seems to be empty in those cases. So, it didn't cause any issue.

`$ mvp --check-inconsistency emits.move` is nothing to worry about because `cargo test` does not do the inconsistency test for the functional tests.

Reply:
The problem is that after you check "expected to fail" in extract_verification_errors, you do not continue to parse the trace and augmented trace. So they remain as unrecognized output. What should be done is move the check to the end of extraction such that the full assert_failed block is parsed. Then you should be able to remove the ad-hoc treatment you added in your PR.